torchchat/generate.py

```diff
@@ -517,26 +572,37 @@ def generate(
 
         # create an empty tensor of the expected final shape and
         # fill in the current tokens
+        if len(prompt.shape) > 1:
+            prompt = prompt.squeeze(0)
         T = prompt.size(0)
         max_new_tokens = min(max_new_tokens, max_seq_length - start_pos - T)
         T_new = T + max_new_tokens
         # set up caches only if first inference
         if start_pos == 0:
             model = model.to(device=device)
             with torch.device(device):
-                if self.is_torchtune_model:
+                if (
+                    self.is_torchtune_model
+                    or self.model.config.model_type == ModelType.Flamingo
+                ):
                     model.setup_caches(max_batch_size=1, dtype=self.dtype)
                 else:
                     model.setup_caches(max_batch_size=1, max_seq_length=max_seq_length)
                 if is_speculative and draft_model is not model:
                     draft_model.setup_caches(
-                        max_batch_size=1, max_seq_length=max_seq_length
+                        max_batch_size=1,
+                        max_seq_length=max_seq_length,
+                        dtype=torch.bfloat16,
```

Comment:
```suggestion
```

Debug remnant

Reply:
setup_caches rn of draft model should always be setup_caches(max_batch_size, max_seq_length), will not have dtype